website/docs/r/kubernetes_cluster_extension.html.markdown

```diff
@@ -0,0 +1,121 @@
+  extension_type      = "microsoft.flux"
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name which should be used for this Kubernetes Cluster Extension. It must be between 1 and 253 characters in length and may contain only letters, numbers, periods (.), hyphens (-), and must begin with a letter or number. Changing this forces a new Kubernetes Cluster Extension to be created.
+
+* `resource_group_name` - (Required) Specifies the name of the Resource Group where the Kubernetes Cluster Extension should exist. Changing this forces a new Kubernetes Cluster Extension to be created.
+
+* `cluster_name` - (Required) Specifies the Cluster Name. Changing this forces a new Kubernetes Cluster Extension to be created.
+
+* `extension_type` - (Required) Type of the Extension, of which this resource is an instance of. It must be one of the Extension Types registered with Microsoft.KubernetesConfiguration by the Extension publisher. For more information, please refer to [Available Extensions for AKS](https://learn.microsoft.com/en-us/azure/aks/cluster-extensions?tabs=azure-cli#currently-available-extensions). Changing this forces a new Kubernetes Cluster Extension to be created.
+
+* `configuration_protected_settings` - (Optional) Configuration settings that are sensitive, as name-value pairs for configuring this extension.
+
+* `configuration_settings` - (Optional) Configuration settings, as name-value pairs for configuring this extension.
+
+* `plan` - (Optional) A `plan` block as defined below.
+
+* `release_train` - (Optional) The release train which is used by this extension. Possible values include but not limited to `Stable`, `Preview`. Changing this forces a new Kubernetes Cluster Extension to be created.
+
+* `release_namespace` - (Optional) Namespace where the extension Release must be placed, for a Cluster scoped extension.  If this namespace does not exist, it will be created. Changing this forces a new Kubernetes Cluster Extension to be created.
+
+* `target_namespace` - (Optional) Namespace where the extension will be created for a Namespace scoped extension.  If this namespace does not exist, it will be created. Changing this forces a new Kubernetes Cluster Extension to be created.
+
+* `version` - (Optional) User-specified version of the extension for this extension to 'pin'. If it is not set, Azure will use the latest version and auto upgrade it. Changing this forces a new Kubernetes Cluster Extension to be created.
+
+---
+
+A `plan` block supports the following:
+
+* `name` - (Required) A user defined name of the 3rd Party Artifact that is being procured. Changing this forces a new Kubernetes Cluster Extension to be created.
+
+* `product` - (Required) Specifies the 3rd Party artifact that is being procured. It maps to the OfferID of Data Market. Changing this forces a new Kubernetes Cluster Extension to be created.
```

Comment:
The `plan` is the same with that in `azurerm_managed_application` so that I leverage the [same description](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/website/docs/r/managed_application.html.markdown#argument-reference).